hrjobcontract/CRM/Hrjobcontract/BAO/HRJobContractRevision.php

```diff
@@ -15,7 +15,9 @@ class CRM_Hrjobcontract_BAO_HRJobContractRevision extends CRM_Hrjobcontract_DAO_
   public static function create($params) {
     global $user;
 
-    $params['editor_uid'] = $user->uid;
+    if (!empty($user->uid)) {
```

Comment:
is this check really needed , the user cannot reach here if he is not logged-in and have the permissions needed .. and if he is logged-in then for sure he have an ID so i feel it is unnecessary.